config/routes.rb

```diff
@@ -123,6 +123,10 @@
       :as => :edit_cart_item
   # delete '/cart/empty' => 'application#empty_cart', :as => :empty_cart
   # put '/cart/update' => 'application#update_cart', :as => :update_cart
+  post '/edit_reservation_form' => 'catalog#submit_cart_updates_form',
+       :as => :submit_cart_updates_form
+  put '/change_reservation_dates' => 'catalog#change_reservation_dates',
+      :as => :change_reservation_dates
 
```

Comment:
Let me clarify the questions - I think we were confused since you don't appear to use these routes anywhere except for in your specs (e.g. `/reload_catalog_cart` instead of `/catalog/reload_catalog_cart`). All the forms where you're specifying the route to PUT to use the `url_for` helper that specifies both a controller and an action (e.g. `/catalog/reload_catalog_cart`), which will be caught by the generalized route matcher [below](https://github.com/YaleSTC/reservations/pull/1221/files#diff-21497849d8f00507c9c8dcaf6288b136L162). Does that make more sense?